vpr/src/route/route_path_manager.cpp

```diff
@@ -0,0 +1,146 @@
+#include "route_path_manager.h"
+#include "globals.h"
+
+PathManager::PathManager() {
+    // Only init data structure if required by RCV
+    // Is disabled by default
+    is_enabled_ = false;
+}
+
+PathManager::~PathManager() {
+    if (is_enabled_) free_all_memory();
```

Comment:
This likely should probably not be behind the `is_enabled_`, because `free_all_memory` should be no-op in the disabled case.

Reply:
True yeah I am doing the is_enabled comparison twice